### ISSUE TITLE
`Query.reply()` should return a `ZFuture`

### ISSUE THREAD

**Issue by OlivierHecart** [eclipse-zenoh/zenoh#251]
`Query.try_reply()` and `Query.reply_async()` should be removed.
`Query.reply()` should return a `ReplyBuilder` which implements `ZFuture` and so provide `.await`, `.wait()` and `.poll()`

**Comment by Mallets** [comment]
59d6b90306c377982e45c2941b935a5efddf436a has been merged into api-changes branch.
This issue will need to be closed manually once api-changes is merged into master.

**Comment by Mallets** [comment]
Merged into master.
